src/crimson/common/gated.h

```diff
@@ -42,7 +53,12 @@ class Gated {
     });
   }
 
-  seastar::future<> close() {
+  template <typename Func>
+  auto submit_with_gate(const char* what, Func&& func) {
+    return seastar::with_gate(pending_dispatch, std::forward<Func>(func));
+  }
+
+  virtual seastar::future<> close() {
```

Comment:
IIUC this shouldn't be virtual, right?

Reply:
yes, after removing the close from gate_per_shard, we don't need it as virtual